daemon/api.go

```diff
@@ -413,13 +431,26 @@ func getSnapsInfo(c *Command, r *http.Request) Response {
 
 		remoteRepo := newRemoteRepo()
 
+		var auther store.Authenticator
+		overlord := c.d.overlord
+		state := overlord.State()
+		state.Lock()
+		user, err := UserFromRequest(state, r)
+		state.Unlock()
+		if err == nil {
+			auther = user.Authenticator()
+		}
+		if err != nil && err != errNoAuth {
+			return InternalError("%v", err)
+		}
```

Comment:
This logic is all exactly the same between the two paths. Can we please transform this into:
```
auther, err := c.d.auther(req)
if err != nil {
        return err
}